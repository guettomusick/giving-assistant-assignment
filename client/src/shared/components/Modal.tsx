import React, { FC } from 'react';
import styled from 'styled-components';

import CloseButton from './CloseButton';
import useStopPropagation from '../hooks/useStopPropagation';

const Backdrop = styled.div`
  position: fixed;
  height: 100vh;
  width: 100vw;
  background-color: rgba(0,0,0,0.3);
  padding: 10vh 10vw;
  cursor: pointer;

  @media(max-width: 768px) {
    padding: 0;
  }
`;

const ModalContainer = styled.div`
  position: relative;
  height: 100%;
  width: 100%;
  background-color: white;
  cursor: default;

  .close-button {
    position: absolute;
    top: 10px;
    right: 10px;
  }
`;

export type Props = {
  open: boolean,
  onClose: () => any,
  onBackdrop: () => any,
  className?: string,
};

const Modal: FC<Props> = ({
  children,
  open,
  onClose,
  onBackdrop,
  className,
}) => {
  const handleClickContainer = useStopPropagation();

  if (!open) {
    return null;
  }

  const modalContainer = (
    <ModalContainer onClick={ handleClickContainer } className={ className }>
      { onClose && <CloseButton className='close-button' onClick={ onClose } /> }
      { children }
    </ModalContainer>
  );

  return onBackdrop
    ? <Backdrop onClick={ onBackdrop }>{ modalContainer }</Backdrop>
    : modalContainer;
};

export default Modal;