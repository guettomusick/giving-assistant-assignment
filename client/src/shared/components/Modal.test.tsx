import React from 'react';
import renderer from 'react-test-renderer';

import Modal, { Props } from './Modal';

describe('Modal', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      open: false,
      onClose: () => {},
      onBackdrop: () => {},
    };
  });

  it('renders closed', () => {
    const component = renderer.create(
      <Modal { ...props }/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders open', () => {
    props.open = true;

    const component = renderer.create(
      <Modal { ...props }/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
