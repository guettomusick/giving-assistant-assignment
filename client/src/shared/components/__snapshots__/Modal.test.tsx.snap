// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders closed 1`] = `null`;

exports[`Modal renders open 1`] = `
<div
  className="sc-AxirZ kXlyGJ"
  onClick={[Function]}
>
  <div
    className="sc-AxiKw jVgpAJ"
    onClick={[Function]}
  >
    <div
      className="sc-AxjAm joHQsR close-button"
      onClick={[Function]}
    >
      <svg
        height="34"
        viewBox="0 0 34 34"
        width="34"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M25.4558 25.4559C30.1421 20.7696 30.1421 13.1716 25.4558 8.4853C20.7696 3.79901 13.1716 3.79901 8.48528 8.4853C3.79899 13.1716 3.79899 20.7696 8.48528 25.4559C13.1716 30.1422 20.7696 30.1422 25.4558 25.4559ZM9.19239 12.0208C8.80186 11.6303 8.80186 10.9971 9.19239 10.6066L10.6066 9.19241C10.9971 8.80188 11.6303 8.80188 12.0208 9.19241L16.9706 14.1422L21.9203 9.19241C22.3108 8.80188 22.944 8.80188 23.3345 9.19241L24.7487 10.6066C25.1393 10.9971 25.1393 11.6303 24.7487 12.0208L19.799 16.9706L24.7487 21.9203C25.1393 22.3109 25.1393 22.944 24.7487 23.3345L23.3345 24.7488C22.944 25.1393 22.3108 25.1393 21.9203 24.7488L16.9706 19.799L12.0208 24.7488C11.6303 25.1393 10.9971 25.1393 10.6066 24.7488L9.19239 23.3345C8.80186 22.944 8.80186 22.3109 9.19239 21.9203L14.1421 16.9706L9.19239 12.0208Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
